libwallet/assets/btc/dex-wallet.go

```diff
@@ -225,17 +225,27 @@ func (dw *DEXWallet) Birthday() time.Time {
 }
 
 func (dw *DEXWallet) Peers() ([]*asset.WalletPeer, error) {
-	peers := dw.spvService.CS.Peers()
-	var walletPeers []*asset.WalletPeer
-	for i := range peers {
-		p := peers[i]
-		walletPeers = append(walletPeers, &asset.WalletPeer{
-			Addr:      p.Addr(),
-			Connected: p.Connected(),
-			Source:    asset.WalletDefault,
-		})
+	peerChan := make(chan []*asset.WalletPeer)
+	go func() {
+		peers := dw.spvService.CS.Peers()
+		var walletPeers []*asset.WalletPeer
+		for i := range peers {
+			p := peers[i]
+			walletPeers = append(walletPeers, &asset.WalletPeer{
+				Addr:      p.Addr(),
+				Connected: p.Connected(),
+				Source:    asset.WalletDefault,
+			})
+		}
+		peerChan <- walletPeers
+	}()
+
+	select {
+	case <-time.After(1 * time.Second):
+		return nil, nil // okay.
+	case peers := <-peerChan:
+		return peers, nil
```

Comment:
The problem is in `*btcChainService.Peers()`; that's where this is needed.
Also, that means that `*DEXWallet.Peers()` should call `spvService.Peers()`, not `spvService.CS.Peers()` but doesn't look like that will be a straightforward change.